src/otlp.rs

```diff
@@ -136,6 +136,16 @@ fn infer_protocol_and_endpoint(
     maybe_protocol: Option<&str>,
     maybe_endpoint: Option<&str>,
 ) -> (String, String) {
+    let maybe_protocol = match maybe_protocol {
+        Some("grpc") => Some("grpc"),
+        Some("http") | Some("http/protobuf") => Some("http/protobuf"),
```

Comment:
According to [OTEL spec](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/#otel_exporter_otlp_protocol), valid values for protocol are:

 * `grpc`
 * `http/protobuf `
 * `http/json`

But our [get_otel_lambda_extension_env_vars](https://github.com/nentgroup/mtg-cloudformation/blob/e06f36cfe649a63f88ee5b1bb7ae446c462ef1d7/components/service/observability.py#L337-L368) helper sets it to "http".

Reply:
@spurge perhaps we should fix this in `mtg-cloudformation`, but then we have to be sure we won't break telemetry for existing go lambdas.